fe/fe-core/src/main/java/com/starrocks/analysis/GrantRoleStmt.java

```diff
@@ -0,0 +1,73 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+package com.starrocks.analysis;
+
+
+import com.starrocks.catalog.Catalog;
+import com.starrocks.cluster.ClusterNamespace;
+import com.starrocks.common.AnalysisException;
+import com.starrocks.common.ErrorCode;
+import com.starrocks.common.ErrorReport;
+import com.starrocks.common.FeNameFormat;
+import com.starrocks.common.UserException;
+import com.starrocks.mysql.privilege.Auth;
+import com.starrocks.mysql.privilege.PrivPredicate;
+import com.starrocks.qe.ConnectContext;
+
+// GRANT Role 'role' TO 'user'
+public class GrantRoleStmt extends DdlStmt {
+    protected String role;
+    protected String qualifiedRole = null;
+    protected UserIdentity userIdent;
+
+    public GrantRoleStmt(String role, UserIdentity userIdent) {
+        this.role = role;
+        this.userIdent = userIdent;
+    }
+
+    public UserIdentity getUserIdent() {
+        return userIdent;
+    }
+
+    public String getQualifiedRole() {
+        return qualifiedRole;
+    }
+
+    @Override
+    public void analyze(Analyzer analyzer) throws AnalysisException, UserException {
```

Comment:
The new Analyzer framework should be used

Reply:
Hi, I'm a newcomer, not sure what "the new Analyzer framework" is. Could you please give me an example of a class using the new Analyzer framework that you've mentioned?